webui/react/src/pages/TaskList.tsx

```diff
@@ -88,6 +93,36 @@ const TaskList: React.FC = () => {
     return false;
   }, [ selectedTasks ]);
 
+  const fetchTasks = useCallback((): void => {
+    requestCommands({});
+    requestNotebooks({});
+    requestShells({});
+    requestTensorboards({});
+  }, [ requestCommands, requestNotebooks, requestShells, requestTensorboards ]);
+
+  /*
+   * Check once every second to see if all task endpoints have resolved.
```

Comment:
it's unfortunate we can't actually expose and wait on the promises for those requests (Promise.all). it might be worthwhile to look into that later.